editoast/src/views/infra/mod.rs

```diff
@@ -543,15 +541,25 @@ async fn get_switch_types(
     })
     .await?;
 
-    let infra = InfraCache::get_or_load(conn, &infra_caches, &infra).await?;
-    Ok(Json(
-        infra
-            .switch_types()
-            .values()
-            .map(ObjectCache::unwrap_switch_type)
-            .cloned()
-            .collect(),
-    ))
+    let selection_settings =
+        SelectionSettings::new().filter(move || SwitchTypeModel::INFRA_ID.eq(infra.id));
+    let switch_types_model = SwitchTypeModel::list(conn, selection_settings)
+        .await
+        .expect("Failed to retrieve extended switch types from the database");
+
+    let extended_switch_types: Vec<_> = switch_types_model
+        .into_iter()
+        .map(SwitchType::from)
+        .collect();
+
+    let builtin_switch_types: Vec<_> = builtin_node_types_list();
```

Comment:
```suggestion
    let builtin_switch_types = builtin_node_types_list();
```

inferred by the compiler from the function's return type